### PR TITLE
#merge the branch with refactored codes with development branch

### PR DESCRIPTION
**Pull request by freemile2ru**
## What does this PR do?
this PR merges the branch with refactored codes with development branch

## Description of Task to be completed?
every file was touched and codes were refactored to meet acceptable standard

## How should this be manually tested?
This can be tested by viewing file changes on this PR

## Any background context you want to provide?
NIL
## What are the relevant pivotal tracker stories?
NIL
## Screenshots (if appropriate)
NIL
##Questions:
NIL

### CODE REVIEW

**Comment by victor-enogwe** [comment]
hello, no description of what you did on these pull request to make it easier to review

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1 @@
+public/js/events.js
```
Comment (victor-enogwe) [inline .eslintignore]: add jquery to eslint and lint this file

File: .eslintrc [hunk 2]
```diff
@@ -1,43 +1,54 @@
...
- },
- "parserOptions": {
-  "ecmaVersion": 6,
-  "sourceType": "module",
+  "root": true,
+  "extends": "airbnb-base",
+  "env": {
+    "node": true,
+    "es6": true,
+    "mocha": true
```
Comment (victor-enogwe) [inline .eslintrc]: you don't use mocha on this project, change the eslint rule mocha true to one that complies with your testing framework

File: karma.conf.js [hunk 3]
```diff
@@ -18,8 +18,8 @@ module.exports = (config) => {
       'https://ajax.googleapis.com/ajax/libs/angularjs/1.6.1/angular.min.js',
       'https://code.angularjs.org/1.6.1/angular-route.min.js',
       'public/js/helper.js',
```
Comment (victor-enogwe) [inline karma.conf.js:20]: use file patterns to load files coming from the same directory and exclude the one you don't want loaded

File: package.json [hunk 4]
```diff
@@ -16,7 +16,6 @@
   "bugs": {
     "url": "https://github.com/andela-uibrahim/inverted-index/issues"
   },
-
   "engines": {
```
Comment (victor-enogwe) [inline package.json]: why specify node version and npm version in your app?

File: public/css/styles.css [hunk 5]
```diff
@@ -96,4 +96,12 @@
 }
```
Comment (victor-enogwe) [inline public/css/styles.css:96]: css rules are not properly aligned

File: public/index.html [hunk 6]
```diff
@@ -10,8 +10,8 @@
 <link rel="stylesheet" href="css/styles.css"/>
 </head>
 
-<body ng-app="myApp">
-<div class="container-fluid blue">
+<body ng-app="myApp" class="blue">
+<div class="container-fluid">
 	<div class="row" id="top">
```
Comment (victor-enogwe) [inline public/index.html:15]: Fix empty divs in this file and use css rules to create all desired visual effects
lint your html properly

File: public/js/InvertedIndexUI/controller.js [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+   */
+    const validateSearchInput = () => {
+      if (!$scope.searchWords) {
+        $scope.alerts(true, 'no search word entered');
+        return false;
+      } else if (!$scope.selected) {
+        $scope.alerts(true, 'select file to search words from');
+        return false;
+      }
+      if ($scope.indexedFiles === undefined) {
```
Comment (victor-enogwe) [inline public/js/InvertedIndexUI/controller.js]: refactor if ($scope.indexedFiles === undefined) to if ($scope.indexedFiles)

File: public/js/events.js [hunk 8]
```diff
@@ -8,5 +8,5 @@ $('.process').click(() => {
   $('.modal-x').trigger('click');
 });
 $('#btn').click(() => {
-	$('#close').trigger('click');
+  $('#close').trigger('click');
```
[the remark this answers is not in the document]
Comment (victor-enogwe) [reply public/js/events.js]: arrow function not properly implemented, remove curly braces from functions that return only one statement.

File: public/js/helper.js [hunk 9]
```diff
@@ -1,97 +1,119 @@
...
+   * @param  {array} book - book to validate content
+   * @return {boolean} result- true or false
+   */
+    validFileContent: (book) => {
+      let result = true;
+      if (!(book instanceof Array)) {
+        return false;
+      }
+      book.forEach((doc) => {
+        if ((doc.title === undefined) || (doc.text === undefined)) {
```
Comment (victor-enogwe) [inline public/js/helper.js]: use negation instead of undefined

File: public/templates/search.html [hunk 10]
```diff
@@ -6,7 +6,7 @@ <h3 class="panel-title">The Inverted Index</h3>
 		</div>
```
Comment (victor-enogwe) [inline public/templates/search.html:6]: lint your html file

File: spec/InvertedIndex.spec.js [hunk 11]
```diff
@@ -0,0 +1,292 @@
...
+   });
+
+   it('should return " array " for a valid json file input', () => {
+     expect((helpers.filterBookContents(book1)) instanceof Array).toBeTruthy();
+   });
+
+   it('should return "an array of books with filtered contents"' +
+   ' when it is invoked with a book', () => {
+     expect(helpers.filterBookContents(book1)).toEqual(
+       [['alice', 'enters', 'a'], ['fellowship', 'wizard', 'on'],
```
Comment (victor-enogwe) [inline spec/InvertedIndex.spec.js]: you can put each arrays on separate lines

**Review by victor-enogwe — CHANGES_REQUESTED** [review]

File: .eslintignore (hunk 1 above)
Comment (freemile2ru) [reply .eslintignore]: @andela-venogwe  i have implemented this feedback

File: karma.conf.js (hunk 3 above)
Comment (freemile2ru) [reply karma.conf.js:20]: @andela-venogwe  please can you shed more light on this?

Comment (victor-enogwe) [reply karma.conf.js:20]: okay, for files in the same directory you can use pattern matching to load them. eg. all files in './public' which have the js extension can be loaded like this './public/*.js'

File: gulpfile.js [hunk 12]
```diff
@@ -1,5 +1,8 @@
 const gulp = require('gulp');
 const browserSync = require('browser-sync');
+const browserify = require('browserify');
+const source = require('vinyl-source-stream');
+const Server = require('karma').Server;
```
Comment (iAmao) [inline gulpfile.js:5]: Good job!

File: public/js/InvertedIndexUI/controller.js [hunk 13]
```diff
@@ -0,0 +1,170 @@
...
+
+   /** reads and load file contents;
+   *
+   * @param  {Array} currentFile - passes the content in current file
+   * @param  {Array} contents - an array that stores the contents
+   * that has been read
+   * @return {Array} array of documents in book
+   */
+    const updateFiles = currentFile => new Promise((resolve) => {
+      const reader = new FileReader();
```
Comment (iAmao) [inline public/js/InvertedIndexUI/controller.js]: I like that you used promises here! Nice one!

File: public/templates/searchContent.html [hunk 14]
```diff
@@ -1,18 +1,21 @@
...
+		<tbody class="table-content">
+			<tr ng-repeat="(word, wordArr) in  searchResult">
+				<td class="text-center">{{word}}</td>
+				<td class="text-center"><span ng-class="{'glyphicon glyphicon-ok' : wordArr[0] === true , 'glyphicon glyphicon-remove' : wordArr[0] === false}"></span></td>
+				<td class="text-center"><span ng-class="{'glyphicon glyphicon-ok' : wordArr[1] === true , 'glyphicon glyphicon-remove' : wordArr[1] === false}"></span></td>
+				<td class="text-center"><span ng-class="{'glyphicon glyphicon-ok' : wordArr[2] === true , 'glyphicon glyphicon-remove' : wordArr[2] === false}"></span></td>
+			</tr>
+		</tbody>
+	</table>
+</div>
```
Comment (iAmao) [inline public/templates/searchContent.html]: I think your html should not be an exception to the indentation rules.

File: spec/InvertedIndex.spec.js [hunk 15]
```diff
@@ -0,0 +1,292 @@
...
+
+   it('should return false when invoked with' +
+   ' an invalid data format', () => {
+     expect(helpers.validFileContent(book2)).toBe(false);
+     expect(helpers.validFileContent(book3)).toBe(false);
+     expect(helpers.validFileContent(book4)).toBe(false);
+   });
+   it('should return false when invoked with' +
+   ' an empty book', () => {
+     expect(helpers.validFileContent([{}])).toBe(false);
```
[the remark this answers is not in the document]
Comment (iAmao) [reply spec/InvertedIndex.spec.js]: There are ways to write the specs without needing line breaks. Also it's not descriptive enough.
For example `should return boolean when invoked with +   ' a valid json file input` could be written as `should return true when passed a valid json file`

**Review by iAmao — APPROVED** [review]

**Pull request merged**